extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/ParquetColumnResolver.java

```diff
@@ -0,0 +1,74 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.parquet.table.location;
+
+import io.deephaven.annotations.BuildableStyle;
+import io.deephaven.engine.table.impl.locations.TableKey;
+import io.deephaven.parquet.table.ParquetInstructions;
+import org.apache.parquet.column.ColumnDescriptor;
+import org.apache.parquet.schema.MessageType;
+import org.immutables.value.Value;
+
+import java.util.Map;
+
+/**
+ * A mapping between Deephaven column names and Parquet {@link ColumnDescriptor column descriptors}.
+ *
+ * TODO: describe better
+ */
+@Value.Immutable
+@BuildableStyle
+public abstract class ParquetColumnResolver {
+
+    /**
+     * {@link ParquetInstructions.Builder#setColumnResolverFactory(Factory)}
+     */
+    public interface Factory {
+
+        /**
+         * TODO: description
+         *
+         * @param tableKey the table key
+         * @param tableLocationKey the Parquet TLK
+         * @return the Parquet column resolver
+         */
+        ParquetColumnResolver init(TableKey tableKey, ParquetTableLocationKey tableLocationKey);
+    }
+
+    public static Builder builder() {
+        return ImmutableParquetColumnResolver.builder();
+    }
+
+    // Intentionally not exposed, but necessary to expose to Builder for safety checks.
+    abstract MessageType schema();
+
+    /**
+     * TODO: javadoc
+     * 
+     * @return
+     */
+    public abstract Map<String, ColumnDescriptor> mapping();
```

Comment:
Should `ParquetColumnResolver` be an interface, with this `Map`-backed implementation as the only one right now?